src/formula/callable.hpp

```diff
@@ -172,11 +154,20 @@ class formula_callable
 
 	TYPE type_;
 
-	mutable std::set<callable_die_subscriber*> dtor_notify;
-
 private:
+	virtual formula_callable* clone() const = 0;
+	const_formula_callable_ptr to_ptr() const
+	{
+		if(weak_from_this().use_count() > 0) {
+			return shared_from_this();
+		}
+		return const_formula_callable_ptr(clone());
```

Comment:
This looks like a source of obscure bugs.

If the caller has a pointer to this, why does it want to keep a pointer instead of just computing the result now? I think the only answer to that is "because something will set more variables later, which may change the value" - therefore silently making a copy here means that something will get the wrong value.

Reply:
There shouldn't be anything setting values on a callable in the middle of formula evaluation (other than temporary callables created as part of the evaluation), so I don't think it's _likely_ that this is a problem, but I'm not able to rule it out, unfortunately.

Ideally there would be no way to construct a formula_callable that's _not_ tracked by a shared_ptr, but that gets quite tricky…